.github/workflows/auto-merge-backports.yml

```diff
@@ -0,0 +1,69 @@
+name: Auto-Merge Test Backport PRs
+
+on:
+  schedule:
+    - cron: "0 * * * *"  # Every hour
+  workflow_dispatch:
+
+jobs:
+  auto-merge:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Merge qualifying PRs
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            // Get timestamp for 24 hours ago in ISO format
+            const now = new Date();
+            const iso24HoursAgo = new Date(now.getTime() - 24 * 60 * 60 * 1000).toISOString();
+
+            const query = `repo:${context.repo.owner}/${context.repo.repo} is:pr is:open label:backport label:blathers-backport label:backport-test-only created:<${iso24HoursAgo}`;
+            const searchResults = await github.paginate(github.rest.search.issuesAndPullRequests, {
+              q: query,
+              per_page: 100,
+            });
+
+            for (const prItem of searchResults) {
+              const prNumber = prItem.number;
+
+              // Fetch full PR details
+              const { data: pr } = await github.rest.pulls.get({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                pull_number: prNumber,
+              });
+
+              // Check for approvals
+              const reviews = await github.rest.pulls.listReviews({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                pull_number: prNumber,
+              });
+
+              const approved = reviews.data.some(r => r.state === 'APPROVED');
+              if (!approved) {
+                console.log(`Skipping PR #${prNumber}: not approved`);
+                continue;
+              }
```

Comment:
In the PR body, it says:

> This action relies on Blathers to label and approve PRs that meet the criteria for auto-merging.

This code only checks for _some_ approval. Not for a _blathers_ approval. Without the check for a blathers approval, it's possible for someone to set the right labels on a PR and have it auto merge, whether intentionally or not. It seems safer to only auto-merge PRs explicitly approved by Blathers (so someone doesn't mistakenly set the label and have their PR merged when they want it open for longer).

Why not have this also verify the aprover is Blathers? Is the intention to have this loophole in there?

Reply:
Updated to check for blathers approval